vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/grid/it/dataview/AbstractItemCountGridPage.java

```diff
@@ -218,7 +218,15 @@ protected void switchToDataProvider() {
     }
 
     protected void switchToDefinedSize() {
-        grid.getLazyDataView().setItemCountCallback(dataProvider::size);
+        grid.getLazyDataView().setItemCountCallback(query -> {
```

Comment:
Another option could have been to change the data provider filter to be `Object` instead of `Void` ?

Now I realize the hindrance, luckily the only one, of not having the filter type parameter in `LazyDataView`. It makes this `itemCountCallback` method a bit useless for both `Grid` and for `ComboBox`. 

Since keeping it makes sense for switching between defined/undefined size without a full reset, I wonder if we should just implement it in the concrete LazyDataView classes for the components, so that the type could go right. It would have some code duplication, but then again the type would be immediately the correct one for the users.